pulsar/internal/lookup_service.go

```diff
@@ -371,8 +396,33 @@ func (h *httpLookupService) GetTopicsOfNamespace(namespace string, mode GetTopic
 	return topics, nil
 }
 
-func (h *httpLookupService) GetSchema(_ string, _ []byte) (schema *pb.Schema, err error) {
-	return nil, errors.New("GetSchema is not supported by httpLookupService")
+func (h *httpLookupService) GetSchema(topic string, schemaVersion []byte) (*LookupSchema, error) {
+	topicName, err := ParseTopicName(topic)
+	if err != nil {
+		return nil, err
+	}
+	topicRestPath := fmt.Sprintf("%s/%s", topicName.Namespace, topicName.Topic)
+	var path string
+	if schemaVersion != nil {
+		path = fmt.Sprintf(HTTPSchemaWithVersionV2, topicRestPath, int64(binary.BigEndian.Uint64(schemaVersion)))
+	} else {
+		path = fmt.Sprintf(HTTPSchemaV2, topicRestPath)
+	}
+	httpSchema := &LookupSchema{}
+	if err := h.httpClient.Get(path, &httpSchema, nil); err != nil {
+		if strings.HasPrefix(err.Error(), "Code: 404") {
+			//	Unlike other language pulsar sdks,
+			//	golang sdk directly use pb.Schema{} binary command instead of lookup service
+			//	in consumer/producer schema initialization.
+			//
+			//	Golang sdk only use this GetSchema() interface in tableView schema query and msg deserialize,
+			//	so that we don't need to take more care about 404 NotFound http exception.
```

Comment:
This comment may not be necessary. The logic is the same as in the Java client, which also ignores the "schema not found" exception.
https://github.com/apache/pulsar/blob/702c0b3cfa251636099871f65f0ba1ac1a52069c/pulsar-client/src/main/java/org/apache/pulsar/client/impl/HttpLookupService.java#L239-L240

Reply:
fixed